### PR TITLE
Remove media collection bridge functions

### PR DESCRIPTION
**Pull request by fluiddot**
**Related PRs:**
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6691

## What?

Remove React Native bridge functions related to media collection.

## Why?

These functions are no longer used, so we are removing them to clean up the bridge code.

## How?

Remove bridge functions and related code.

## Testing Instructions

The changes shouldn't affect the current version of the editor. For testing, we could just smoke testing the editor and observe that works as expected.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

File: packages/react-native-bridge/index.js [hunk 1]
```diff
@@ -262,16 +262,6 @@ export function mediaUploadSync() {
...
 
-/**
- * Request to start listening to save events when in-progress saves are in place
- *
- * For example, when media is being saved and the user re-enters the editor
- *
- */
-export function mediaSaveSync() {
-	return RNReactNativeGutenbergBridge.mediaSaveSync();
-}
```
Comment (fluiddot) [inline packages/react-native-bridge/index.js:273]: Only used in Story block ([reference](https://github.com/Automattic/jetpack/pull/36151/files#diff-3ec47af4510e547616f9dbcdcd2557a8cd539dcaf938c8b1c1a7d5f42fc696a7L43)).

File: packages/react-native-bridge/index.js [hunk 2]
```diff
@@ -352,79 +342,6 @@ export function showXpostSuggestions() {
...
-/**
- * Request the host app to show the block for editing its mediaFiles collection
- *
- * For example, a mediaFiles collection editor can make special handling of visualization
- * in this regard.
- *
- * @param {Array<Map>} mediaFiles    the mediaFiles attribute of the block, containing data about each media item.
- * @param {string}     blockClientId the clientId of the block.
- */
-export function requestMediaFilesEditorLoad( mediaFiles, blockClientId ) {
```
Comment (fluiddot) [inline packages/react-native-bridge/index.js:364]: Only used in Story block ([reference](https://github.com/Automattic/jetpack/pull/36151/files#diff-3ec47af4510e547616f9dbcdcd2557a8cd539dcaf938c8b1c1a7d5f42fc696a7L51-L54))

File: packages/react-native-bridge/index.js [hunk 3]
```diff
@@ -352,79 +342,6 @@ export function showXpostSuggestions() {
...
-/**
- * Request the host app to show a retry dialog for mediaFiles arrays which contained items that failed
- * to upload
- *
- * For example, tapping on a failed-media overlay would trigger this request and a "Retry?" dialog
- * would be presented to the user
- *
- * @param {Array<Map>} mediaFiles the mediaFiles attribute of the block, containing data about each media item
- */
-export function requestMediaFilesFailedRetryDialog( mediaFiles ) {
```
Comment (fluiddot) [inline packages/react-native-bridge/index.js:380]: Only used in Story block ([reference](https://github.com/Automattic/jetpack/pull/36151/files#diff-3ec47af4510e547616f9dbcdcd2557a8cd539dcaf938c8b1c1a7d5f42fc696a7L164))

File: packages/react-native-bridge/index.js [hunk 4]
```diff
@@ -352,79 +342,6 @@ export function showXpostSuggestions() {
...
-
-/**
- * Request the host app to show a cancel dialog for mediaFiles arrays currently being uploaded
- *
- * For example, tapping on a block containing mediaFiles that are currently being uplaoded would trigger this request
- * and a "Cancel upload?" dialog would be presented to the user.
- *
- * @param {Array<Map>} mediaFiles the mediaFiles attribute of the block, containing data about each media item
- */
-export function requestMediaFilesUploadCancelDialog( mediaFiles ) {
```
Comment (fluiddot) [inline packages/react-native-bridge/index.js:394]: Only used in Story block ([reference](https://github.com/Automattic/jetpack/pull/36151/files#diff-3ec47af4510e547616f9dbcdcd2557a8cd539dcaf938c8b1c1a7d5f42fc696a7L160))

File: packages/react-native-bridge/index.js [hunk 5]
```diff
@@ -352,79 +342,6 @@ export function showXpostSuggestions() {
...
-
-/**
- * Request the host app to show a cancel dialog for mediaFiles arrays currently undergoing a save operation
- *
- * Save operations on mediaFiles collection could  be lengthy so for example, tapping on a mediaFiles-type block
- * currently being saved would trigger this request and a "Cancel save?" dialog would be presented to the user
- *
- * @param {Array<Map>} mediaFiles the mediaFiles attribute of the block, containing data about each media item.
- */
-export function requestMediaFilesSaveCancelDialog( mediaFiles ) {
```
Comment (fluiddot) [inline packages/react-native-bridge/index.js:408]: Only used in Story block ([reference](https://github.com/Automattic/jetpack/pull/36151/files#diff-3ec47af4510e547616f9dbcdcd2557a8cd539dcaf938c8b1c1a7d5f42fc696a7L162))

File: packages/react-native-bridge/index.js [hunk 6]
```diff
@@ -352,79 +342,6 @@ export function showXpostSuggestions() {
...
-}
-
-/**
- * Request the host app to listen to mediaFiles collection based block replacement signals
- * in case such an event was enqueued
- *
- * @param {Array<Map>} mediaFiles    the mediaFiles attribute of the block, containing data about each media item.
- * @param {string}     blockClientId the clientId of the block.
- */
-export function mediaFilesBlockReplaceSync( mediaFiles, blockClientId ) {
```
Comment (fluiddot) [inline packages/react-native-bridge/index.js:421]: Only used in Story block ([reference](https://github.com/Automattic/jetpack/pull/36151/files#diff-3ec47af4510e547616f9dbcdcd2557a8cd539dcaf938c8b1c1a7d5f42fc696a7L45-L49))

File: packages/react-native-bridge/ios/GutenbergBridgeDelegate.swift [hunk 7]
```diff
@@ -294,13 +280,4 @@ public extension GutenbergBridgeDelegate {
...
     func gutenbergDidSendButtonPressedAction(_ buttonType: Gutenberg.ActionButtonType) { }
-
-    // Media Collection
-
-    func gutenbergDidRequestMediaSaveSync() {}
-    func gutenbergDidRequestMediaFilesEditorLoad(_ mediaFiles: [[String: Any]], blockId: String) { }
-    func gutenbergDidRequestMediaFilesFailedRetryDialog(_ mediaFiles: [[String: Any]]) { }
-    func gutenbergDidRequestMediaFilesUploadCancelDialog(_ mediaFiles: [[String: Any]]) { }
-    func gutenbergDidRequestMediaFilesSaveCancelDialog(_ mediaFiles: [[String: Any]]) { }
-    func gutenbergDidRequestMediaFilesBlockReplaceSync(_ mediaFiles: [[String: Any]], clientId: String) {}
```
Comment (fluiddot) [inline packages/react-native-bridge/ios/GutenbergBridgeDelegate.swift:305]: These functions are optional, only `gutenbergDidRequestMediaFilesEditorLoad` is actually implemented in the host app.

**Review by SiobhyB — APPROVED** [review]
LGTM, smoke tested the editor on both platforms and confirmed I spotted no regressions. 🚀

**Pull request merged**